javascriptlib/src/mill/javascriptlib/TscModule.scala

```diff
@@ -0,0 +1,730 @@
+package mill.javascriptlib
+
+import mill.*
+import os.*
+
+import scala.annotation.tailrec
+import scala.util.Try
+
+trait TscModule extends Module { outer =>
```

Comment:
Why the rename?

Reply:
It was discussed at the start of #3927, I never got around to doing it.
Plus this is much easier to type out.

It does not provide any real value other than easy to type out makes extension of the Module look more compact

Reply:
@monyedavid got it. Let's keep the old name `TypeScriptModule` for now, it matches the other module naming conventions, and the incidental rename makes this PR impossible to review

Reply:
Done @lihaoyi